Remove a BUG() call that would crash on a race condition that should
otherwise be harmless.

--- a/drivers/mtd/nand/pxa3xx_nand.c
+++ b/drivers/mtd/nand/pxa3xx_nand.c
@@ -701,7 +701,6 @@ static void handle_data_pio(struct pxa3x
 	default:
 		dev_err(&info->pdev->dev, "%s: invalid state %d\n", __func__,
 				info->state);
-		BUG();
 	}
 
 	/* Update buffer pointers for multi-page read/write */
